vm/src/builtins/zip.rs

```diff
@@ -1,15 +1,19 @@
 use super::PyTypeRef;
 use crate::{
-    function::PosArgs,
+    builtins::IntoPyBool,
+    function::{OptionalArg, PosArgs},
     protocol::{PyIter, PyIterReturn},
     slots::{IteratorIterable, SlotConstructor, SlotIterator},
-    PyClassImpl, PyContext, PyRef, PyResult, PyValue, VirtualMachine,
+    IntoPyObject, PyClassImpl, PyContext, PyObjectRef, PyRef, PyResult, PyValue, TryFromObject,
+    TypeProtocol, VirtualMachine,
 };
+use crossbeam_utils::atomic::AtomicCell;
 
 #[pyclass(module = false, name = "zip")]
 #[derive(Debug)]
 pub struct PyZip {
     iterators: Vec<PyIter>,
+    strict: AtomicCell<bool>,
```

Comment:
Use a `PyAtomic<bool>` here. To get the same semantics as `AtomicCell` use `Acquire` ordering on loads and `Release` on stores.

Reply:
Doesn't use atomics when threading is disabled.